app/models/agents/tumblr_likes_agent.rb

```diff
@@ -0,0 +1,58 @@
+module Agents
+  class TumblrLikesAgent < Agent
+    include TumblrConcern
+
+    gem_dependency_check { defined?(Tumblr::Client) }
+
+    description <<-MD
+      The Tumblr Likes Agent checks for liked Tumblr posts from a specific blog.
+
+      #{'## Include `tumblr_client` and `omniauth-tumblr` in your Gemfile to use this Agent!' if dependencies_missing?}
+
+      To be able to use this Agent you need to authenticate with Tumblr in the [Services](/services) section first.
+
+
+      **Required fields:**
+
+      `blog_name` The Tumblr URL you're querying (e.g. "staff.tumblr.com")
+
+      Set `expected_update_period_in_days` to the maximum amount of time that you'd expect to pass between Events being created by this Agent.
+    MD
+
+    default_schedule 'every_1h'
+
+    def validate_options
+      errors.add(:base, 'blog_name is required') unless options['blog_name'].present?
+      errors.add(:base, 'expected_update_period_in_days is required') unless options['expected_update_period_in_days'].present?
+    end
+
+    def working?
+      event_created_within?(interpolated['expected_update_period_in_days']) && !recent_error_logs?
+    end
+
+    def default_options
+      {
+        'expected_update_period_in_days' => '10',
+        'blog_name' => '{{blog_name}}',
+      }
+    end
+
+    def check
+      liked = tumblr.blog_likes(interpolated['blog_name'])
+      memory[:ids] ||= []
+
+      if liked['liked_posts']
+        # Loop over all liked posts which came back from Tumblr, add to memory, and create events.
+        liked['liked_posts'].each do |post|
+          unless memory[:ids].include?(post['id'])
+            memory[:ids].push(post['id'])
```

Comment:
It looks like nothing ever cleans up this array from `memory`. In theory it could get quite large, and `include` is searching it for every `post` in `liked`, so it could become a performance issue. I'd suggest something like this (I haven't tested it):

```ruby
memory[:ids] ||= []
if liked['liked_posts']
  # Loop over all liked posts which came back from Tumblr, add to memory, and create events.
  lookup = memory[:ids].each.with_object({}) { |id, obj| obj[id.to_s] = true }
  liked['liked_posts'].each do |post|
    unless lookup[post['id'].to_s]
      memory[:ids].push(post['id'])
      create_event(payload: post)
    end
  end

  max_lookback =  (options['max_lookback'].presence || MAX_LOOKBACK).to_i
  if memory[:ids].length > max_lookback
    memory[:ids] = memory[:ids].last(max_lookback)
  end
elsif ...
```